lib/ClangImporter/ImportType.cpp

```diff
@@ -2478,6 +2479,19 @@ static ParamDecl *getParameterInfo(ClangImporter::Implementation *impl,
   paramInfo->setInterfaceType(swiftParamTy);
   impl->recordImplicitUnwrapForDecl(paramInfo, isParamTypeImplicitlyUnwrapped);
 
+  // Swift doesn't support default values of inout parameters.
+  if (param->hasDefaultArg() && !isInOut &&
+      impl->isCxxInteropCompatVersionAtLeast(
+          version::getUpcomingCxxInteropCompatVersion())) {
+    SwiftDeclSynthesizer synthesizer(*impl);
+    if (CallExpr *defaultArgExpr = synthesizer.makeDefaultArgument(
+            param, swiftParamTy, paramInfo->getParameterNameLoc())) {
+      paramInfo->setDefaultArgumentKind(DefaultArgumentKind::Normal);
+      paramInfo->setDefaultExpr(defaultArgExpr, /*isTypeChecked*/ true);
+      paramInfo->setDefaultValueStringRepresentation("cxxDefaultArg");
```

Comment:
Another idea - since using a specific string value to represent the default value in the module interface, can you instead render the original Clang expression and present it to the user as a sort of special expression , e.g. `foo(x: Foo = #cxxDefaultArg(Foo(1))`

Reply:
That is smart, I like this idea!